src/rdkafka_telemetry_decode.c

```diff
@@ -0,0 +1,543 @@
+                                               const uint8_t *decoded) {
+        switch (unit_test_data.current_field) {
+        case 2:
+                rd_snprintf(unit_test_data.metric_name,
+                            sizeof(unit_test_data.metric_name), "%s", decoded);
+                break;
+        case 3:
+                rd_snprintf(unit_test_data.metric_description,
+                            sizeof(unit_test_data.metric_description), "%s",
+                            decoded);
+                break;
+        default:
+                break;
+        }
+        unit_test_data.current_field++;
+}
+
+static void unit_test_telemetry_decoded_NumberDataPoint(
+    void *opaque,
+    const opentelemetry_proto_metrics_v1_NumberDataPoint *decoded) {
+        unit_test_data.metric_value_int    = decoded->value.as_int;
+        unit_test_data.metric_value_double = decoded->value.as_double;
+        unit_test_data.metric_time         = decoded->time_unix_nano;
+        unit_test_data.current_field++;
+}
+
+static void
+unit_test_telemetry_decoded_type(void *opaque,
+                                 rd_kafka_telemetry_metric_type_t type) {
+        unit_test_data.type = type;
+        unit_test_data.current_field++;
+}
+
+static void unit_test_telemetry_error(void *opaque, const char *error, ...) {
+        char buffer[1024];
+        va_list ap;
+        va_start(ap, error);
+        rd_vsnprintf(buffer, sizeof(buffer), error, ap);
+        va_end(ap);
+        RD_UT_SAY("%s", buffer);
+}
+
+bool unit_test_telemetry(rd_kafka_telemetry_producer_metric_name_t metric_name,
+                         const char *expected_name,
+                         const char *expected_description,
+                         rd_kafka_telemetry_metric_type_t expected_type,
+                         rd_bool_t is_double) {
+        rd_kafka_t *rk = rd_calloc(1, sizeof(*rk));
+        rwlock_init(&rk->rk_lock);
+        rk->rk_type                          = RD_KAFKA_PRODUCER;
+        rk->rk_telemetry.matched_metrics_cnt = 1;
+        rk->rk_telemetry.matched_metrics =
+            rd_malloc(sizeof(rd_kafka_telemetry_producer_metric_name_t) *
+                      rk->rk_telemetry.matched_metrics_cnt);
+        rk->rk_telemetry.matched_metrics[0] = metric_name;
+        rd_strlcpy(rk->rk_name, "unittest", sizeof(rk->rk_name));
+        clear_unit_test_data();
+
+        rd_kafka_telemetry_decode_interface_t decode_interface = {
+            .decoded_string = unit_test_telemetry_decoded_string,
+            .decoded_NumberDataPoint =
+                unit_test_telemetry_decoded_NumberDataPoint,
+            .decoded_int64 = NULL,
+            .decoded_type  = unit_test_telemetry_decoded_type,
+            .decode_error  = unit_test_telemetry_error,
+            .opaque        = &unit_test_data,
+        };
+
+        TAILQ_INIT(&rk->rk_brokers);
+
+        rd_kafka_broker_t *rkb  = rd_calloc(1, sizeof(*rkb));
+        rkb->rkb_c.connects.val = 1;
```

Comment:
Add an initialization here, this will fix the issue on mingw when calling `rd_avg_destroy` because of the call to `mtx_destroy` with a mutex not initialized.
```suggestion
        rkb->rkb_c.connects.val = 1;
        rd_avg_init(&rkb->rkb_telemetry.rd_avg_current.rkb_avg_rtt,
                    RD_AVG_GAUGE, 0, 500 * 1000, 2, rd_true);
        rd_avg_init(&rkb->rkb_telemetry.rd_avg_current.rkb_avg_outbuf_latency,
                    RD_AVG_GAUGE, 0, 500 * 1000, 2, rd_true);
        rd_avg_init(&rkb->rkb_telemetry.rd_avg_current.rkb_avg_throttle,
                    RD_AVG_GAUGE, 0, 500 * 1000, 2, rd_true);
        rd_avg_init(&rkb->rkb_telemetry.rd_avg_rollover.rkb_avg_rtt,
                    RD_AVG_GAUGE, 0, 500 * 1000, 2, rd_true);
        rd_avg_init(&rkb->rkb_telemetry.rd_avg_rollover.rkb_avg_outbuf_latency,
                    RD_AVG_GAUGE, 0, 500 * 1000, 2, rd_true);
        rd_avg_init(&rkb->rkb_telemetry.rd_avg_rollover.rkb_avg_throttle,
                    RD_AVG_GAUGE, 0, 500 * 1000, 2, rd_true);

```

Reply:
Added